### PR TITLE
Update mkl to 2020.2.254

### LINKED ISSUE #52907: Windows builds are using Microsoft's OpenMP header instead of Intel's
## 🐛 Bug

Windows builds on CI seem to be using [Microsoft's OpenMP](https://docs.microsoft.com/en-us/cpp/build/reference/openmp-enable-openmp-2-0-support?view=msvc-160) headers instead of Intel's. Is it possible to use Intel's OpenMP instead, as [Microsoft OpenMP doesn't support all the functions](https://docs.microsoft.com/en-us/cpp/parallel/openmp/reference/openmp-functions?view=msvc-160) in the OpenMP specification, such as [omp_get_level()](https://www.openmp.org/spec-html/5.0/openmpsu127.html). The build process installs `intel-openmp`, which is being used, but as @peterjc123 astutely observed, the omp.h header file provided by Microsoft is being used.  Thanks!

```
-- Found OpenMP_C: -openmp:experimental  
CMake Warning (dev) at C:/w/b/windows/conda/envs/py37/Library/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:426 (message):
  The package name passed to `find_package_handle_standard_args` (OpenMP_CXX)
  does not match the name of the calling package (OpenMP).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/Modules/FindOpenMP.cmake:565 (find_package_handle_standard_args)
  third_party/fbgemm/CMakeLists.txt:60 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.


-- Found OpenMP_CXX: -openmp:experimental  
-- Found OpenMP: TRUE   
CMake Warning at third_party/fbgemm/CMakeLists.txt:62 (message):
  OpenMP found! OpenMP_C_INCLUDE_DIRS =
```

```
CXX flags             : /DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -DNDEBUG -DUSE_FBGEMM -DUSE_XNNPACK
```

## To Reproduce

1. Use a standard OpenMP function that Microsoft doesn't list in its [ OpenMP functions list](https://docs.microsoft.com/en-us/cpp/parallel/openmp/reference/openmp-functions?view=msvc-160), such as [omp_get_level()](https://www.openmp.org/spec-html/5.0/openmpsu127.html).
2. Windows builds will fail, as [here](https://app.circleci.com/pipelines/github/pytorch/pytorch/278273/workflows/297e8ae2-c210-466c-be24-ee5c82787d2f/jobs/11168659), for example, because Intel OpenMP's headers aren't used.
3. #52964 by @peterjc123 resolves this issue.

## Expected behavior

Is it possible to use Intel's OpenMP headers for Windows builds instead? 

## Environment

[CI](https://app.circleci.com/pipelines/github/pytorch/pytorch/278273/workflows/297e8ae2-c210-466c-be24-ee5c82787d2f/jobs/11168659)

## Additional context
#52964 by @peterjc123 resolves this issue.

### PR DESCRIPTION
**Pull request by peterjc123**
Fixes https://github.com/pytorch/pytorch/issues/52907

### CODE REVIEW

**Comment by imaginary-person** [comment]
Thanks a ton, @peterjc123!

File: .circleci/scripts/binary_checkout.sh [hunk 1]
```diff
@@ -61,7 +61,7 @@ git --no-pager log --max-count 1
 popd
 
 # Clone the Builder master repo
-retry git clone -q https://github.com/pytorch/builder.git "$BUILDER_ROOT"
+retry git clone -q https://github.com/peterjc123/builder.git -b mkl_update "$BUILDER_ROOT"
```
Comment (peterjc123) [inline .circleci/scripts/binary_checkout.sh:64]: Note: Undo this change before merge

File: cmake/Modules/FindOpenMP.cmake [hunk 2]
```diff
@@ -132,6 +132,7 @@ set(OpenMP_C_CXX_TEST_SOURCE
 int main(void) {
 #ifdef _OPENMP
   omp_get_max_threads();
+  omp_get_level();
```
Comment (peterjc123) [inline cmake/Modules/FindOpenMP.cmake:135]: Note: Undo this change before merge

**Review by seemethere — APPROVED** [review]

**Comment by malfet** [comment]
Have we considered potential perf degradation on AMD CPUs? See https://github.com/pytorch/builder/issues/504

**Comment by imaginary-person** [comment]
A part of this PR enables the use of Intel OpenMP headers instead of Microsoft's on Windows. Since it'd probably be rolled back, perhaps a separate PR can be created on its basis, just for enabling Intel OpenMP headers, without updating the MKL version. Since @peterjc123 worked on it, I'd wait for him to submit it, if he'd like to. Otherwise, I can do so.

**Comment by peterjc123** [comment]
@malfet If this is a problem, then we probably should downgrade to 2020.0.
